service/history/replication/task_processor_test.go

```diff
@@ -167,6 +172,167 @@ func (s *taskProcessorSuite) TestSendFetchMessageRequest() {
 	s.NotNil(requestMessage.respChan)
 }
 
+func (s *taskProcessorSuite) TestProcessorLoop_RespChanClosed() {
+	// start the process loop
+	s.taskProcessor.wg.Add(1)
+	go s.taskProcessor.processorLoop()
+	defer close(s.taskProcessor.done)
+
+	// act like taskFetcher here and populate respChan of the request
+	requestMessage := <-s.requestChan
+	close(requestMessage.respChan)
+	time.Sleep(50 * time.Millisecond)
+
+	// loop should have continued by now. validate by checking the new request
+	select {
+	case <-s.requestChan:
+	// expected
+	case <-time.After(50 * time.Millisecond):
+		s.Fail("new request not sent to requestChan")
+	}
+}
+
+func (s *taskProcessorSuite) TestProcessorLoop_TaskExecuteSuccess() {
```

Comment:
These tests are more like component tests. Why not use fake functions so that we don't need to worry about the logic in `processResponse` function?

Reply:
I could have gone that route but there's still value in such multi-function coverage. if the test logic became too complicated to mock all downstream calls then using function overrides would make more sense